xsnippet/api/application.py

```diff
@@ -65,12 +70,9 @@ def create_app(conf):
             middlewares.auth.auth(conf['auth']),
         ],
         router=router.VersionRouter({'1.0': v1}, default='1.0'))
-    app.on_startup.append(middlewares.auth.setup)
 
-    # Attach settings to the application instance in order to make them
-    # accessible at any point of execution (e.g. request handling).
-    app['conf'] = conf
-    app.on_startup.append(database.setup)
+    app.on_startup.append(functools.partial(middlewares.auth.setup, conf=conf))
+    app.on_startup.append(functools.partial(database.setup, db=db))
```

Comment:
I wonder if the app should be concerned with the database connection at all, now, that we do not need to carry a reference to a database connection around, but can just consume it from picobox. On the other hand,  we do need to call this `setup()` function from somewhere, so maybe it's just ok.

Reply:
Yeah, I have the very same feeling about this. Honestly, I just don't know yet where it would be better to put, so I decided to go with this for now and think about it a little bit later.

One thing I know for sure, we do not need `middlewares.auth.setup` to be called from here. It generates a secret when it's not set, so I'm thinking we should move it away from `create_app` and maybe put in `__main__`. Besides, doing so will allow us to have `auth` module untied from the app, hence make it reusable.